openwisp_controller/connection/apps.py

```diff
@@ -64,3 +77,53 @@ def _is_update_in_progress(cls, device_pk):
                 if task['name'] == _TASK_NAME and str(device_pk) in task['args']:
                     return True
         return False
+
+    @classmethod
+    def is_working_changed_receiver(cls, instance, is_working, **kwargs):
+        device = instance.device
+        notification_opts = dict(sender=instance, target=device)
+        if not is_working:
+            notification_opts['type'] = 'connection_is_not_working'
+        else:
+            notification_opts['type'] = 'connection_is_working'
+        notify.send(**notification_opts)
+
+    def register_notification_types(self):
+        register_notification_type(
+            'connection_is_not_working',
+            {
+                'verbose_name': 'Device Connection PROBLEM',
+                'verb': 'not working',
+                'level': 'error',
+                'email_subject': (
+                    '[{site.name}] PROBLEM: Connection to '
+                    'device {notification.target}'
+                ),
+                'message': (
+                    '{notification.actor.credentials} connection to '
+                    'device [{notification.target}]({notification.target_link}) '
+                    'is {notification.verb}. {notification.actor.failure_reason}'
+                ),
+            },
+        )
+        register_notification_type(
+            'connection_is_working',
+            {
+                'verbose_name': 'Device Connection RECOVERY',
+                'verb': 'working',
+                'level': 'info',
+                'email_subject': (
+                    '[{site.name}] RECOVERY: Connection to '
+                    'device {notification.target}'
+                ),
+                'message': (
+                    '{notification.actor.credentials} connection to '
+                    'device [{notification.target}]({notification.target_link}) '
+                    'is {notification.verb}. {notification.actor.failure_reason}'
+                ),
+            },
+        )
+        try:
+            unregister_notification_type('default')
+        except ImproperlyConfigured:
+            pass
```

Comment:
watch out, this unregistration is already  being done in `config`: https://github.com/openwisp/openwisp-controller/blob/master/openwisp_controller/config/apps.py#L115-L119
Let's not do it twice.